plugins/wporg-5ftf/includes/stats.php

```diff
@@ -0,0 +1,185 @@
+<?php
+
+/**
+ * Track and display key metrics for the program, to measure growth and effectiveness.
+ */
+
+namespace WordPressDotOrg\FiveForTheFuture\Stats;
+
+use WordPressDotOrg\FiveForTheFuture;
+use WordPressDotOrg\FiveForTheFuture\{ Contributor, Pledge, XProfile };
+use WP_Query;
+
+defined( 'WPINC' ) || die();
+
+add_action( 'init',                 __NAMESPACE__ . '\register_post_types' );
+add_action( 'init',                 __NAMESPACE__ . '\schedule_cron_jobs' );
+add_action( '5ftf_record_snapshot', __NAMESPACE__ . '\record_snapshot' );
+
+add_shortcode( '5ftf_stats', __NAMESPACE__ . '\render_shortcode' );
+
+
+/**
+ * Register the snapshots post type. Each post represents a snapshot of the stats at a point in time.
+ */
+function register_post_types() {
+	$args = array(
+		'supports'     => array( 'custom-fields' ),
+		'public'       => false,
+		'show_in_rest' => true,
+
+		// Only allow posts to be created programmatically.
+		'capability_type' => '5ftf_stats',
+		'capabilities'    => array(
+			'create_posts' => 'do_not_allow',
+		),
+	);
+
+	register_post_type( '5ftf_stats_snapshot', $args );
```

Comment:
I like the pattern in Contributor & Pledge that we set up constants for this, I think it'd be good to do that here too.

https://github.com/WordPress/five-for-the-future/blob/e4deb6809ac70b6a95f3c0cf9157a17fbfe2aeeb/plugins/wporg-5ftf/includes/pledge.php#L17-L19